docs/002-refresh-certs.md

```diff
@@ -0,0 +1,347 @@
+  Seed string `json:"seed"`
+  // ExpirationSeconds is the duration after which the certificates will expire.
+  ExpirationSeconds int `json:"expiration-seconds"`
+  //ExtraSANs is a list of additional Subject Alternative Names to include in the certificate.
+  ExtraSANs []string `json:"extra-sans"`
+}
+```
+
+### `POST /x/capi/approve-certificates`
+
+This endpoint will approve the renewal of certificates for a worker node and
+will be run by a control plane node. The request will include a list of the CSR
+names to approve, recovered from the `k8sd.io/refresh-certificates-csr-names`.
+
+```go
+type ApproveCertificatesRequest struct {
+  // CertificateSigningRequests is a list of CSR names to approve.
+  CertificateSigningRequests []string `json:"certificate-signing-requests"`
+}
+```
+
+## Bootstrap Provider Changes
+<!--
+This section MUST mention any changes to the bootstrap provider.
+-->
+
+A controller called `CertificatesController` will be added to the bootstrap
+provider. This controller will watch for the `k8sd.io/refresh-certificates`
+annotation on the Machine object and trigger the certificate renewal process
+when the annotation is present.
+
+### Control Plane Nodes
+
+The controller would use the value of the `k8sd.io/refresh-certificates` annotation
+to determine the duration after which the certificates will expire. It will then
+call the `POST /x/capi/refresh-certificates` endpoint to trigger the certificate
+renewal process.
+
+The controller will share the status of the certificate renewal process with the
+Machine object by updating the `k8sd.io/refresh-certificates-status` annotation
+with the status of the renewal process. The value of this annotation will be
+one of the following:
+
+- `in-progress`: The certificate renewal process is in progress.
+- `done`: The certificate renewal process is complete.
+- `failed`: The certificate renewal process has failed.
+
+After the certificate renewal process is complete, the controller will update
+the `machine.cluster.x-k8s.io/certificates-expiry` annotation on the Machine
+object with the new expiry date of the certificates.
+
+Finally, the controller will remove the `k8sd.io/refresh-certificates` annotation
+from the Machine object to indicate that the certificate renewal process is
+complete.
+
+### Worker Nodes
+
+The controller would use the value of the `k8sd.io/refresh-certificates` annotation
+to determine the duration after which the certificates will expire. It will then
+call the `POST /x/capi/request-certificates` endpoint to create the Certificate
+Signing Request (CSR) for the worker node.
+
+The controller will share the CSR names with the control plane node by updating
+the `k8sd.io/refresh-certificates-csr-names` annotation with the list of CSR names
+to approve. The control plane node will then call the `POST /x/capi/approve-certificates`
+endpoint to approve the Certificate Signing Requests list provided by the annotation
+value.
```

Comment:
I'm not sure if we need to annotate control plane machines, we have the k8sd proxy which we can use to talk to the k8sd  of a control plane node directly. I assume the provider can just use this proxy to request the CSR's to be approved.